### PR TITLE
Cordova UT app can now refresh content on the fly

### PR DESCRIPTION
**Pull request by shrishrirang**
- The unit test directory is restructured for better code sharing between browser and cordova 
- Added a cordova app for unit testing
- QUnit is added as an npm dependency instead of loading from http://qunit... 
- The Cordova UT app can refresh content on the fly by loading the specified URL

### CODE REVIEW

**Comment by shrishrirang** [comment]
\+@danderson00 for review.
The number of files is large but most are due to a newly added VS Cordova project. Looking at individual commit descriptions should help you look at relevant diffs.

File: Gruntfile.js [hunk 1]
```diff
@@ -119,13 +119,25 @@ module.exports = function(grunt) {
...
+                src: '*',
+                dest: 'sdk/test/app/browser/external/qunit/',
+                expand: true,
+                cwd: 'node_modules/qunitjs/qunit'
+            },
+            cordovaTest: {
+                src: '*',
+                dest: 'sdk/test/app/cordova/www/external/qunit/',
+                expand: true,
+                cwd: 'node_modules/qunitjs/qunit'
```
Comment (danderson00) [inline Gruntfile.js]: Do we have different tests that run on web and cordova? Push maybe?

Comment (shrishrirang) [inline Gruntfile.js]: The push unit tests run on both: browser and Cordova. The sqlite store unit tests in the offline branch run only on Cordova.

Comment (danderson00) [inline Gruntfile.js]: Yeh, no worries. Ideally, we would bundle all of the tests in a single package and filter out the inappropriate tests from each run (e.g. mocha -g offline --invert) but I'm not sure if qunit has this sort of functionality. This is a reasonable mechanism for our scenario.

**Comment by danderson00** [comment]
Looks good, a good enhancement to the workflow.

**Pull request merged**